rlp/sedes/serializable.py

```diff
@@ -205,7 +205,23 @@ def __init__(self, *args, **kwargs):
         for value, attr in zip(field_values, self._meta.field_attrs):
             setattr(self, attr, make_immutable(value))
 
-    _cached_rlp = None
+    _cached_rlp_data = None
+    _cached_serial = None
+
+    @property
+    def _cached_rlp(self):
+        if self._cached_rlp_data:
+            return self._cached_rlp_data
+        elif self._cached_serial:
+            from rlp.codec import encode_raw
+            self._cached_rlp_data = encode_raw(self._cached_serial)
```

Comment:
So this is just re-encoding the string of bytes which represents the object we're decoding?

Reply:
IIUC, `cached_serial` will be a sequence of items that can be passed to `Serializable.deserialize()` to construct the actual object. `encode_raw` will simply join those in a string of bytes, adding the type/length RLP prefixes